Notebooks/notebook_temporary/utilities.py

```diff
@@ -63,3 +74,320 @@ def get_group_id(crop_group, crop_group_df, group_name='GROUP_1_NAME',
         return default_value
     else:
         return values[-1]
+
+
+def get_eopatch_ids(id_bottom_left, id_top_right, bbox_grid):
+    """Generate 2D array of eopatch IDs from bottom-left and top-right eopatch IDs.
+    IDs are identified from eo-patch bounding box grid provided in the form of
+    geopandas data frame.
+
+    :param id_bottom_left: Bottom-left eopatch ID
+    :type id_bottom_left: int
+    :param id_top_right: Top-right eopatch ID
+    :type id_top_right: int
+    :param bbox_grid: Grid of eopatch bounding boxes in Geopandas data frame format
+    :type bbox_grid: DataFrame
+    :return: 2D array of Eopatch IDs
+    :rtype: numpy.array
+    """
+    coords_bl = list(bbox_grid.loc[id_bottom_left, ['index_x', 'index_y']])
+    coords_tr = list(bbox_grid.loc[id_top_right, ['index_x', 'index_y']])
+
+    ids = []
+    for y in range(coords_tr[1], coords_bl[1] - 1, -1):
+        row = []
+        for x in range(coords_bl[0], coords_tr[0] + 1):
+            indices = list(bbox_grid[bbox_grid['index_x'] == x][bbox_grid['index_y'] == y].index)
+            row.append(indices[0] if len(indices) else None)
+        ids.append(row)
+
+    return np.array(ids)
+
+
+def plot_grid(id_grid, patch_dir, img_func, img_func_args={}, imshow_args={}, title=None, colorbar=None, size=20):
+    """Plot grid of eopatches defined by grid of eopatch IDs and plotting function.
+
+    :param id_grid: 2D array of Eopatch IDs
+    :type id_grid: numpy.array
+    :param patch_dir: Path to the directory where eopatches are stored.
+        Eopatch directory names should be formated as `eopatch_N`, where `N` is eopatch ID.
+    :type patch_dir: string
+    :param img_func: Function that prepares image data from eopatch
+    :type img_func: function
+    :param img_func_args: `func` arguments, defaults to {}
+    :type img_func_args: dict, optional
+    :param imshow_args: `imshow_args` arguments, defaults to {}
+    :type imshow_args: dict, optional
+    :param title: Suptitle, defaults to None
+    :type title: str, optional
+    :param colorbar: Colorbar settings, i.e. `ticks` and `labels`, defaults to None
+    :type colorbar: dict or None, optional
+    :param size: Plot size in inches
+    :type size: int
+    """
+    grid_shape = id_grid.shape
+    fig, axs = plt.subplots(nrows=grid_shape[0], ncols=grid_shape[1])
+    patch_dirs = [os.path.join(patch_dir, f'eopatch_{patch_id}') for patch_id in id_grid.ravel()]
```

Comment:
The code would be better to go over all the patches in a folder, regardless of the name. Something like this
`os.walk(patch_dir)`
And the positions could be determined from the bboxes.